Android app force close

**Describe the issue**
Android force closing every few seconds, after running in the background for an hour.

**To Reproduce or Feature details**
Steps to reproduce the behavior:
1. Open the app
2. Keep it on the background
3. Keep doing other stuffs on your phone like browsing web.
4. See error

**Expected behavior**
No force close

**Screenshots**
 
![unnamed (2)](https://user-images.githubusercontent.com/20724062/87062242-6eb51e80-c22a-11ea-88a0-f6b0b81c073f.jpg)


**Smartphone (please complete as much of the following information as possible):
Device: Asus max pro m1
OS: Android 9

**Additional context**
Maybe add a log collector in the app.

The issue which causes app close is as follows:

`1 {"LOGCAT":"07-24 02:10:26.753 E\/AndroidRuntime( 3297): FATAL EXCEPTION: main\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): Process: org.amahi.anywhere, PID: 3297\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): java.lang.RuntimeException: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.amahi.anywhere\/.service.UploadService }: app is in background uid UidRecord{46b8a0 u0a602 TRNB bg:+29m52s456ms idle change:uncached procs:2 seq(0,0,0)}\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat android.app.job.JobServiceEngine$JobHandler.handleMessage(JobServiceEngine.java:112)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat android.os.Handler.dispatchMessage(Handler.java:106)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat android.os.Looper.loop(Looper.java:192)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat android.app.ActivityThread.main(ActivityThread.java:6754)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat java.lang.reflect.Method.invoke(Native Method)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:549)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:828)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.amahi.anywhere\/.service.UploadService }: app is in background uid UidRecord{46b8a0 u0a602 TRNB bg:+29m52s456ms idle change:uncached procs:2 seq(0,0,0)}\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat android.app.ContextImpl.startServiceCommon(ContextImpl.java:1525)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat android.app.ContextImpl.startService(ContextImpl.java:1481)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat android.content.ContextWrapper.startService(ContextWrapper.java:650)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat org.amahi.anywhere.job.NetConnectivityJob.onStartJob(NetConnectivityJob.java:79)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat android.app.job.JobService$1.onStartJob(JobService.java:71)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \tat android.app.job.JobServiceEngine$JobHandler.handleMessage(JobServiceEngine.java:108)\n07-24 02:10:26.753 E\/AndroidRuntime( 3297): \t... 6 more\n07-24 02:10:26.756 E\/ACRA ( 3297): ACRA caught a RuntimeException for org.amahi.anywhere\n07-24 02:10:26.756 E\/ACRA ( 3297): java.lang.RuntimeException: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.amahi.anywhere\/.service.UploadService }: app is in background uid UidRecord{46b8a0 u0a602 TRNB bg:+29m52s456ms idle change:uncached procs:2 seq(0,0,0)}\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat android.app.job.JobServiceEngine$JobHandler.handleMessage(JobServiceEngine.java:112)\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat android.os.Handler.dispatchMessage(Handler.java:106)\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat android.os.Looper.loop(Looper.java:192)\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat android.app.ActivityThread.main(ActivityThread.java:6754)\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat java.lang.reflect.Method.invoke(Native Method)\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:549)\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:828)\n07-24 02:10:26.756 E\/ACRA ( 3297): Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.amahi.anywhere\/.service.UploadService }: app is in background uid UidRecord{46b8a0 u0a602 TRNB bg:+29m52s456ms idle change:uncached procs:2 seq(0,0,0)}\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat android.app.ContextImpl.startServiceCommon(ContextImpl.java:1525)\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat android.app.ContextImpl.startService(ContextImpl.java:1481)\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat android.content.ContextWrapper.startService(ContextWrapper.java:650)\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat org.amahi.anywhere.job.NetConnectivityJob.onStartJob(NetConnectivityJob.java:79)\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat android.app.job.JobService$1.onStartJob(JobService.java:71)\n07-24 02:10:26.756 E\/ACRA ( 3297): \tat android.app.job.JobServiceEngine$JobHandler.handleMessage(JobServiceEngine.java:108)\n07-24 02:10:26.756 E\/ACRA ( 3297): \t... 6 more\n07-24 02:10:26.923 I\/ACRA ( 3237): Sending report \/data\/user\/0\/org.amahi.anywhere\/app_ACRA-approved\/2020-07-24T02:10:05.462+05:30.stacktrace\n07-24 02:10:27.039 I\/FtBuild ( 3297): getRomVersion mRomVer=4.5\n07-24 02:10:27.049 V\/Typeface( 3297): CurFontPath: \/system\/fonts\/Roboto-Regular.ttf\n07-24 02:10:27.049 D\/Typeface( 3297): switch to android orgin ttf\n07-24 02:10:27.082 D\/OpenGLRenderer( 3297): HWUI GL Pipeline\n07-24 02:10:27.821 I\/Adreno ( 3297): QUALCOMM build : 99230b5, I19c5914ec0\n07-24 02:10:27.821 I\/Adreno ( 3297): Build Date : 12\/03\/18\n07-24 02:10:27.821 I\/Adreno ( 3297): OpenGL ES Shader Compiler Version: EV031.22.00.01_06_07\n07-24 02:10:27.821 I\/Adreno ( 3297): Local Branch : \n07-24 02:10:27.821 I\/Adreno ( 3297): Remote Branch : \n07-24 02:10:27.821 I\/Adreno ( 3297): Remote Branch : \n07-24 02:10:27.821 I\/Adreno ( 3297): Reconstruct Branch : \n07-24 02:10:27.831 I\/Adreno ( 3297): PFP: 0x005ff087, ME: 0x005ff063\n07-24 02:10:27.846 I\/zygote64( 3297): android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 0\n07-24 02:10:27.847 I\/OpenGLRenderer( 3297): Initialized EGL, version 1.4\n07-24 02:10:27.847 D\/OpenGLRenderer( 3297): Swap behavior 2\n07-24 02:10:28.034 E\/RenderScript( 3297): SETAFFINITY ret = -1\n07-24 02:10:28.034 E\/RenderScript( 3297): SETAFFINITY ret = -1\n07-24 02:10:28.034 E\/RenderScript( 3297): SETAFFINITY ret = -1\n07-24 02:10:28.034 E\/RenderScript( 3297): SETAFFINITY ret = -1\n07-24 02:10:28.034 E\/RenderScript( 3297): SETAFFINITY ret = -1\n07-24 02:10:28.034 E\/RenderScript( 3297): SETAFFINITY ret = -1\n07-24 02:10:28.035 E\/RenderScript( 3297): SETAFFINITY ret = -1\n07-24 02:10:29.132 I\/Process ( 3297): Sending signal. PID: 3297 SIG: 9\n07-24 02:10:29.241 I\/ACRA ( 3237): Sending report \/data\/user\/0\/org.amahi.anywhere\/app_ACRA-approved\/2020-07-24T02:10:26.778+05:30.stacktrace\n07-24 02:10:36.731 I\/zygote64( 3745): Late-enabling -Xcheck:jni\n07-24 02:10:36.944 I\/ACRA ( 3745): ACRA is enabled for org.amahi.anywhere, initializing...\n07-24 02:10:37.891 I\/PhotosContentJob( 3745): JOB STARTED!\n07-24 02:10:37.907 E\/JobServiceEngine( 3745): Error while executing job: 125\n07-24 02:10:37.908 D\/AndroidRuntime( 3745): Shutting down VM\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): FATAL EXCEPTION: main\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): Process: org.amahi.anywhere, PID: 3745\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): java.lang.RuntimeException: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.amahi.anywhere\/.service.UploadService (has extras) }: app is in background uid UidRecord{46b8a0 u0a602 TRNB bg:+30m3s638ms idle change:uncached procs:2 seq(0,0,0)}\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat android.app.job.JobServiceEngine$JobHandler.handleMessage(JobServiceEngine.java:112)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat android.os.Handler.dispatchMessage(Handler.java:106)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat android.os.Looper.loop(Looper.java:192)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat android.app.ActivityThread.main(ActivityThread.java:6754)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat java.lang.reflect.Method.invoke(Native Method)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:549)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:828)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.amahi.anywhere\/.service.UploadService (has extras) }: app is in background uid UidRecord{46b8a0 u0a602 TRNB bg:+30m3s638ms idle change:uncached procs:2 seq(0,0,0)}\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat android.app.ContextImpl.startServiceCommon(ContextImpl.java:1525)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat android.app.ContextImpl.startService(ContextImpl.java:1481)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat android.content.ContextWrapper.startService(ContextWrapper.java:650)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat org.amahi.anywhere.job.PhotosContentJob.onStartJob(PhotosContentJob.java:108)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat android.app.job.JobService$1.onStartJob(JobService.java:71)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \tat android.app.job.JobServiceEngine$JobHandler.handleMessage(JobServiceEngine.java:108)\n07-24 02:10:37.913 E\/AndroidRuntime( 3745): \t... 6 more\n07-24 02:10:37.916 E\/ACRA ( 3745): ACRA caught a RuntimeException for org.amahi.anywhere\n07-24 02:10:37.916 E\/ACRA ( 3745): java.lang.RuntimeException: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.amahi.anywhere\/.service.UploadService (has extras) }: app is in background uid UidRecord{46b8a0 u0a602 TRNB bg:+30m3s638ms idle change:uncached procs:2 seq(0,0,0)}\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat android.app.job.JobServiceEngine$JobHandler.handleMessage(JobServiceEngine.java:112)\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat android.os.Handler.dispatchMessage(Handler.java:106)\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat android.os.Looper.loop(Looper.java:192)\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat android.app.ActivityThread.main(ActivityThread.java:6754)\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat java.lang.reflect.Method.invoke(Native Method)\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:549)\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:828)\n07-24 02:10:37.916 E\/ACRA ( 3745): Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.amahi.anywhere\/.service.UploadService (has extras) }: app is in background uid UidRecord{46b8a0 u0a602 TRNB bg:+30m3s638ms idle change:uncached procs:2 seq(0,0,0)}\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat android.app.ContextImpl.startServiceCommon(ContextImpl.java:1525)\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat android.app.ContextImpl.startService(ContextImpl.java:1481)\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat android.content.ContextWrapper.startService(ContextWrapper.java:650)\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat org.amahi.anywhere.job.PhotosContentJob.onStartJob(PhotosContentJob.java:108)\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat android.app.job.JobService$1.onStartJob(JobService.java:71)\n07-24 02:10:37.916 E\/ACRA ( 3745): \tat android.app.job.JobServiceEngine$JobHandler.handleMessage(JobServiceEngine.java:108)\n07-24 02:10:37.916 E\/ACRA ( 3745): \t... 6 more","USER_CRASH_DATE":"2020-07-24T02:10:37.937+05:30","STACK_TRACE":"java.lang.RuntimeException: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.amahi.anywhere\/.service.UploadService (has extras) }: app is in background uid UidRecord{46b8a0 u0a602 TRNB bg:+30m3s638ms idle change:uncached procs:2 seq(0,0,0)}\n\tat android.app.job.JobServiceEngine$JobHandler.handleMessage(JobServiceEngine.java:112)\n\tat android.os.Handler.dispatchMessage(Handler.java:106)\n\tat android.os.Looper.loop(Looper.java:192)\n\tat android.app.ActivityThread.main(ActivityThread.java:6754)\n\tat java.lang.reflect.Method.invoke(Native Method)\n\tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:549)\n\tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:828)\nCaused by: java.lang.IllegalStateException: Not allowed to start service Intent { cmp=org.amahi.anywhere\/.service.UploadService (has extras) }: app is in background uid UidRecord{46b8a0 u0a602 TRNB bg:+30m3s638ms idle change:uncached procs:2 seq(0,0,0)}\n\tat android.app.ContextImpl.startServiceCommon(ContextImpl.java:1525)\n\tat android.app.ContextImpl.startService(ContextImpl.java:1481)\n\tat android.content.ContextWrapper.startService(ContextWrapper.java:650)\n\tat org.amahi.anywhere.job.PhotosContentJob.onStartJob(PhotosContentJob.java:108)\n\tat android.app.job.JobService$1.onStartJob(JobService.java:71)\n\tat android.app.job.JobServiceEngine$JobHandler.handleMessage(JobServiceEngine.java:108)\n\t... 6 more\n","IS_SILENT":false,"REPORT_ID":"d7ba40cb-7822-4c24-9b94-a376c8c72f49"}`

--- a/src/main/java/org/amahi/anywhere/AmahiApplication.java
+++ b/src/main/java/org/amahi/anywhere/AmahiApplication.java
@@ -25,6 +25,7 @@ import android.app.NotificationManager;
 import android.content.Context;
 import android.content.SharedPreferences;
 import android.os.Build;
+import android.os.StrictMode;
 import android.preference.PreferenceManager;
 
 import androidx.annotation.RequiresApi;
@@ -127,7 +128,7 @@ public class AmahiApplication extends Application {
 
     private void setUpDetecting() {
         if (isDebugging()) {
-//            StrictMode.enableDefaults();
+            StrictMode.enableDefaults();
         }
     }
 

--- a/src/main/java/org/amahi/anywhere/AmahiApplication.java
+++ b/src/main/java/org/amahi/anywhere/AmahiApplication.java
@@ -55,8 +55,8 @@ import timber.log.Timber;
 public class AmahiApplication extends Application {
     private ObjectGraph injector;
 
-    private static final String UPLOAD_CHANNEL_ID = "file_upload";
-    private static final String DOWNLOAD_CHANNEL_ID = "file_download";
+    public static final String UPLOAD_CHANNEL_ID = "file_upload";
+    public static final String DOWNLOAD_CHANNEL_ID = "file_download";
 
     private Boolean isLightThemeEnabled = false;
     private static AmahiApplication instance = null;

--- a/src/main/java/org/amahi/anywhere/AmahiApplication.java
+++ b/src/main/java/org/amahi/anywhere/AmahiApplication.java
@@ -159,12 +159,13 @@ public class AmahiApplication extends Application {
     private void createNotificationChannel() {
 
         // Creating NotificationChannel only for API 26+
-        int importance = NotificationManager.IMPORTANCE_DEFAULT;
+        int importanceDownload = NotificationManager.IMPORTANCE_LOW;
+        int importanceUpload = NotificationManager.IMPORTANCE_LOW;
 
-        NotificationChannel uploadChannel = new NotificationChannel(UPLOAD_CHANNEL_ID, getString(R.string.upload_channel), importance);
+        NotificationChannel uploadChannel = new NotificationChannel(UPLOAD_CHANNEL_ID, getString(R.string.upload_channel), importanceUpload);
         uploadChannel.setDescription(getString(R.string.upload_channel_desc));
 
-        NotificationChannel downloadChannel = new NotificationChannel(DOWNLOAD_CHANNEL_ID, getString(R.string.download_channel), importance);
+        NotificationChannel downloadChannel = new NotificationChannel(DOWNLOAD_CHANNEL_ID, getString(R.string.download_channel), importanceDownload);
         downloadChannel.setDescription(getString(R.string.download_channel_desc));
 
         // Once the channel is registered, it's importance and behaviour can't be changed

--- a/src/main/java/org/amahi/anywhere/AmahiApplication.java
+++ b/src/main/java/org/amahi/anywhere/AmahiApplication.java
@@ -25,7 +25,6 @@ import android.app.NotificationManager;
 import android.content.Context;
 import android.content.SharedPreferences;
 import android.os.Build;
-import android.os.StrictMode;
 import android.preference.PreferenceManager;
 
 import androidx.annotation.RequiresApi;
@@ -178,7 +177,7 @@ public class AmahiApplication extends Application {
         super.attachBaseContext(base);
         if (isDebugging()) {
 
-            if(Api.toSendMail()) {
+            if (Api.toSendMail()) {
                 CoreConfigurationBuilder builder = new CoreConfigurationBuilder(this)
                     .setBuildConfigClass(BuildConfig.class)
                     .setReportFormat(StringFormat.JSON)

--- a/src/main/java/org/amahi/anywhere/AmahiApplication.java
+++ b/src/main/java/org/amahi/anywhere/AmahiApplication.java
@@ -128,7 +128,7 @@ public class AmahiApplication extends Application {
 
     private void setUpDetecting() {
         if (isDebugging()) {
-            StrictMode.enableDefaults();
+//            StrictMode.enableDefaults();
         }
     }
 

--- a/src/main/java/org/amahi/anywhere/AmahiModule.java
+++ b/src/main/java/org/amahi/anywhere/AmahiModule.java
@@ -48,6 +48,7 @@ import org.amahi.anywhere.fragment.UploadSettingsFragment;
 import org.amahi.anywhere.server.ApiModule;
 import org.amahi.anywhere.service.AudioService;
 import org.amahi.anywhere.service.DownloadService;
+import org.amahi.anywhere.service.ServiceNotifier;
 import org.amahi.anywhere.service.UploadService;
 import org.amahi.anywhere.service.VideoService;
 import org.amahi.anywhere.task.AudioMetadataRetrievingTask;
@@ -104,6 +105,7 @@ import dagger.Provides;
         ServerFileTvFragment.class,
         UploadService.class,
         DownloadService.class,
+        ServiceNotifier.class,
         UploadManager.class,
         TvPlaybackVideoFragment.class,
         TvPlaybackVideoActivity.class,

--- a/src/main/java/org/amahi/anywhere/activity/NavigationActivity.java
+++ b/src/main/java/org/amahi/anywhere/activity/NavigationActivity.java
@@ -23,19 +23,20 @@ import android.content.Context;
 import android.content.Intent;
 import android.content.res.Configuration;
 import android.os.Bundle;
-import androidx.annotation.Nullable;
-import androidx.fragment.app.Fragment;
-import androidx.drawerlayout.widget.DrawerLayout;
-import androidx.appcompat.app.ActionBarDrawerToggle;
-import androidx.appcompat.app.AppCompatActivity;
-import androidx.appcompat.app.AppCompatDelegate;
-import androidx.appcompat.widget.Toolbar;
-import androidx.core.view.GravityCompat;
 import android.view.MenuItem;
 import android.view.View;
 import android.view.ViewStub;
 import android.widget.FrameLayout;
 import android.widget.RelativeLayout;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.ActionBarDrawerToggle;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.app.AppCompatDelegate;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.view.GravityCompat;
+import androidx.drawerlayout.widget.DrawerLayout;
+import androidx.fragment.app.Fragment;
 
 import com.squareup.otto.Subscribe;
 
@@ -69,7 +70,7 @@ import javax.inject.Inject;
  * The navigation itself is done via {@link org.amahi.anywhere.fragment.NavigationFragment},
  * {@link org.amahi.anywhere.fragment.ServerSharesFragment} and {@link org.amahi.anywhere.fragment.ServerAppsFragment}.
  */
-    public class NavigationActivity extends AppCompatActivity implements DrawerLayout.DrawerListener {
+public class NavigationActivity extends AppCompatActivity implements DrawerLayout.DrawerListener {
     @Inject
     ServerClient serverClient;
     private ActionBarDrawerToggle navigationDrawerToggle;
@@ -149,7 +150,7 @@ import javax.inject.Inject;
     }
 
     private void hideActionBar() {
-        if (getSupportActionBar()!=null) getSupportActionBar().hide();
+        if (getSupportActionBar() != null) getSupportActionBar().hide();
     }
 
     private void setUpInjections() {

--- a/src/main/java/org/amahi/anywhere/activity/NavigationActivity.java
+++ b/src/main/java/org/amahi/anywhere/activity/NavigationActivity.java
@@ -69,7 +69,7 @@ import javax.inject.Inject;
  * The navigation itself is done via {@link org.amahi.anywhere.fragment.NavigationFragment},
  * {@link org.amahi.anywhere.fragment.ServerSharesFragment} and {@link org.amahi.anywhere.fragment.ServerAppsFragment}.
  */
-public class NavigationActivity extends AppCompatActivity implements DrawerLayout.DrawerListener {
+    public class NavigationActivity extends AppCompatActivity implements DrawerLayout.DrawerListener {
     @Inject
     ServerClient serverClient;
     private ActionBarDrawerToggle navigationDrawerToggle;

--- a/src/main/java/org/amahi/anywhere/activity/RecentFilesActivity.java
+++ b/src/main/java/org/amahi/anywhere/activity/RecentFilesActivity.java
@@ -526,7 +526,11 @@ public class RecentFilesActivity extends AppCompatActivity implements
 
     private void startDownloadService(ServerFile file) {
         Intent downloadService = Intents.Builder.with(this).buildDownloadServiceIntent(file, null);
-        startService(downloadService);
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            startForegroundService(downloadService);
+        } else {
+            startService(downloadService);
+        }
     }
 
     private static final class State {

--- a/src/main/java/org/amahi/anywhere/activity/ServerFileAudioActivity.java
+++ b/src/main/java/org/amahi/anywhere/activity/ServerFileAudioActivity.java
@@ -25,6 +25,7 @@ import android.content.Intent;
 import android.content.ServiceConnection;
 import android.graphics.drawable.Drawable;
 import android.net.Uri;
+import android.os.Build;
 import android.os.Bundle;
 import android.os.IBinder;
 import android.os.Parcelable;
@@ -330,7 +331,11 @@ public class ServerFileAudioActivity extends AppCompatActivity implements
 
     private void setUpAudioService() {
         Intent intent = new Intent(this, AudioService.class);
-        startService(intent);
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            startForegroundService(intent);
+        } else {
+            startService(intent);
+        }
     }
 
     private void setUpAudioServiceBind() {

--- a/src/main/java/org/amahi/anywhere/activity/ServerFileVideoActivity.java
+++ b/src/main/java/org/amahi/anywhere/activity/ServerFileVideoActivity.java
@@ -249,7 +249,11 @@ public class ServerFileVideoActivity extends AppCompatActivity implements
 
     private void setUpVideoService() {
         Intent intent = new Intent(this, VideoService.class);
-        startService(intent);
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            startForegroundService(intent);
+        } else {
+            startService(intent);
+        }
     }
 
     private void setUpVideoServiceBind() {

--- a/src/main/java/org/amahi/anywhere/fragment/ServerFilesFragment.java
+++ b/src/main/java/org/amahi/anywhere/fragment/ServerFilesFragment.java
@@ -423,7 +423,11 @@ public class ServerFilesFragment extends Fragment implements
 
     private void startDownloadService(ServerFile file) {
         Intent downloadService = Intents.Builder.with(getContext()).buildDownloadServiceIntent(file, getShare());
-        getContext().startService(downloadService);
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            getContext().startForegroundService(downloadService);
+        } else {
+            getContext().startService(downloadService);
+        }
     }
 
     private void updateCurrentFileOfflineState(boolean enable) {

--- a/src/main/java/org/amahi/anywhere/job/NetConnectivityJob.java
+++ b/src/main/java/org/amahi/anywhere/job/NetConnectivityJob.java
@@ -50,13 +50,13 @@ public class NetConnectivityJob extends JobService {
         JOB_INFO = builder.build();
     }
 
-    private final String TAG = this.getClass().getName();
+    private static final String TAG = NetConnectivityJob.class.getName();
 
     // Schedule this job, replace any existing one.
     public static void scheduleJob(Context context) {
         JobScheduler js = context.getSystemService(JobScheduler.class);
         js.schedule(JOB_INFO);
-        Log.i("NetworkConnectivityJob", "JOB SCHEDULED!");
+        Log.i(TAG, "JOB SCHEDULED!");
     }
 
     // Check whether this job is currently scheduled.

--- a/src/main/java/org/amahi/anywhere/job/NetConnectivityJob.java
+++ b/src/main/java/org/amahi/anywhere/job/NetConnectivityJob.java
@@ -28,8 +28,9 @@ import android.content.Context;
 import android.content.Intent;
 import android.net.ConnectivityManager;
 import android.os.Build;
-import androidx.annotation.RequiresApi;
 import android.util.Log;
+
+import androidx.annotation.RequiresApi;
 
 import org.amahi.anywhere.AmahiApplication.JobIds;
 import org.amahi.anywhere.service.DownloadService;
@@ -76,7 +77,11 @@ public class NetConnectivityJob extends JobService {
     public boolean onStartJob(JobParameters params) {
         Log.i(TAG, "JOB STARTED!");
         Intent intent = new Intent(this, UploadService.class);
-        startService(intent);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            startForegroundService(intent);
+        } else {
+            startService(intent);
+        }
         startDownloadService(this);
         return false;
     }
@@ -84,7 +89,11 @@ public class NetConnectivityJob extends JobService {
     private void startDownloadService(Context context) {
         Intent downloadService = new Intent(context, DownloadService.class);
         downloadService.setAction(ConnectivityManager.CONNECTIVITY_ACTION);
-        context.startService(downloadService);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            context.startForegroundService(downloadService);
+        } else {
+            context.startService(downloadService);
+        }
     }
 
     @Override

--- a/src/main/java/org/amahi/anywhere/job/PhotosContentJob.java
+++ b/src/main/java/org/amahi/anywhere/job/PhotosContentJob.java
@@ -63,14 +63,14 @@ public class PhotosContentJob extends JobService {
         JOB_INFO = builder.build();
     }
 
-    private final String TAG = this.getClass().getName();
+    private static final String TAG = PhotosContentJob.class.getName();
     JobParameters mRunningParams;
 
     // Schedule this job, replace any existing one.
     public static void scheduleJob(Context context) {
         JobScheduler js = context.getSystemService(JobScheduler.class);
         js.schedule(JOB_INFO);
-        Log.i("PhotosContentJob", "JOB SCHEDULED!");
+        Log.i(TAG, "JOB SCHEDULED!");
     }
 
     // Check whether this job is currently scheduled.
@@ -88,7 +88,7 @@ public class PhotosContentJob extends JobService {
 
     @Override
     public boolean onStartJob(JobParameters params) {
-        Log.i("PhotosContentJob", "JOB STARTED!");
+        Log.i(TAG, "JOB STARTED!");
         mRunningParams = params;
 
         // Did we trigger due to a content change?

--- a/src/main/java/org/amahi/anywhere/job/PhotosContentJob.java
+++ b/src/main/java/org/amahi/anywhere/job/PhotosContentJob.java
@@ -105,7 +105,7 @@ public class PhotosContentJob extends JobService {
                     }
                 }
                 Intent intent = Intents.Builder.with(this).buildUploadServiceIntent(uris);
-                startService(intent);
+                startForegroundService(intent);
             } else {
                 // We don't have any details about URIs (because too many changed at once)
                 Log.i(TAG, "Photos rescan needed!");

--- a/src/main/java/org/amahi/anywhere/receiver/CameraReceiver.java
+++ b/src/main/java/org/amahi/anywhere/receiver/CameraReceiver.java
@@ -22,6 +22,7 @@ package org.amahi.anywhere.receiver;
 import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
+import android.os.Build;
 import android.util.Log;
 
 import org.amahi.anywhere.util.Intents;
@@ -34,6 +35,10 @@ public class CameraReceiver extends BroadcastReceiver {
     public void onReceive(Context context, Intent intent) {
         Log.e("NEW_IMAGE", "onReceive");
         Intent uploadService = Intents.Builder.with(context).buildUploadServiceIntent(intent.getData());
-        context.startService(uploadService);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            context.startForegroundService(uploadService);
+        } else {
+            context.startService(uploadService);
+        }
     }
 }

--- a/src/main/java/org/amahi/anywhere/receiver/NetworkReceiver.java
+++ b/src/main/java/org/amahi/anywhere/receiver/NetworkReceiver.java
@@ -24,6 +24,7 @@ import android.content.Context;
 import android.content.Intent;
 import android.net.ConnectivityManager;
 import android.net.NetworkInfo;
+import android.os.Build;
 
 import org.amahi.anywhere.bus.BusProvider;
 import org.amahi.anywhere.bus.NetworkChangedEvent;
@@ -60,7 +61,11 @@ public class NetworkReceiver extends BroadcastReceiver {
 
     private void startUploadService(Context context) {
         Intent uploadService = new Intent(context, UploadService.class);
-        context.startService(uploadService);
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            context.startForegroundService(uploadService);
+        } else {
+            context.startService(uploadService);
+        }
     }
 
     private void stopUploadService(Context context) {
@@ -71,7 +76,11 @@ public class NetworkReceiver extends BroadcastReceiver {
     private void startDownloadService(Context context) {
         Intent downloadService = new Intent(context, DownloadService.class);
         downloadService.setAction(ConnectivityManager.CONNECTIVITY_ACTION);
-        context.startService(downloadService);
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            context.startForegroundService(downloadService);
+        } else {
+            context.startService(downloadService);
+        }
     }
 
 }

--- a/src/main/java/org/amahi/anywhere/service/DownloadService.java
+++ b/src/main/java/org/amahi/anywhere/service/DownloadService.java
@@ -1,5 +1,6 @@
 package org.amahi.anywhere.service;
 
+import android.app.ActivityManager;
 import android.app.DownloadManager;
 import android.app.Notification;
 import android.app.NotificationManager;
@@ -10,11 +11,12 @@ import android.net.ConnectivityManager;
 import android.net.Uri;
 import android.os.Build;
 import android.os.IBinder;
+import android.util.Log;
+
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.core.app.NotificationCompat;
 import androidx.localbroadcastmanager.content.LocalBroadcastManager;
-import android.util.Log;
 
 import com.squareup.otto.Subscribe;
 
@@ -139,8 +141,23 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
             isDownloading = true;
         } else {
             stopDownloading();
-            stopForeground(true);
+            if (isServiceRunningInForeground(this, DownloadService.class)) {
+                stopForeground(true);
+            }
         }
+    }
+
+    public static boolean isServiceRunningInForeground(Context context, Class<?> serviceClass) {
+        ActivityManager manager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
+        for (ActivityManager.RunningServiceInfo service : manager.getRunningServices(Integer.MAX_VALUE)) {
+            if (serviceClass.getName().equals(service.service.getClassName())) {
+                if (service.foreground) {
+                    return true;
+                }
+
+            }
+        }
+        return false;
     }
 
     private void stopDownloading() {
@@ -151,7 +168,7 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
         offlineFile.setDownloadId(downloadId);
         offlineFileRepository.update(offlineFile);
     }
-
+    
     private void resumeDownload(long downloadId, String fileName) {
         notificationBuilder = new NotificationCompat.Builder(getApplicationContext(), DOWNLOAD_CHANNEL_ID);
         notificationBuilder

--- a/src/main/java/org/amahi/anywhere/service/DownloadService.java
+++ b/src/main/java/org/amahi/anywhere/service/DownloadService.java
@@ -1,6 +1,5 @@
 package org.amahi.anywhere.service;
 
-import android.annotation.SuppressLint;
 import android.app.DownloadManager;
 import android.app.Notification;
 import android.app.NotificationManager;
@@ -48,7 +47,6 @@ import static org.amahi.anywhere.util.Downloader.OFFLINE_PATH;
 public class DownloadService extends ServiceNotifier implements Downloader.DownloadCallbacks {
 
     private static final int NOTIFICATION_OFFLINE_ID = 101;
-    private static final String DOWNLOAD_CHANNEL_ID = "file_download";
 
     @Inject
     ServerClient serverClient;
@@ -74,7 +72,7 @@ public class DownloadService extends ServiceNotifier implements Downloader.Downl
         setUpNetworkUtils();
         setDatabaseRepository();
         downloader.setDownloadCallbacks(this);
-        notificationBuilder = startForegroundNotif(DOWNLOAD_CHANNEL_ID);
+        notificationBuilder = startForegroundNotif(AmahiApplication.DOWNLOAD_CHANNEL_ID);
         Log.i("DownloadService", "Download Service created");
     }
 
@@ -154,7 +152,7 @@ public class DownloadService extends ServiceNotifier implements Downloader.Downl
     }
 
     private void resumeDownload(long downloadId, String fileName) {
-        notificationBuilder = new NotificationCompat.Builder(getApplicationContext(), DOWNLOAD_CHANNEL_ID);
+        notificationBuilder = new NotificationCompat.Builder(getApplicationContext(), AmahiApplication.DOWNLOAD_CHANNEL_ID);
         notificationBuilder
             .setOngoing(true)
             .setSmallIcon(R.drawable.ic_app_logo)
@@ -227,7 +225,7 @@ public class DownloadService extends ServiceNotifier implements Downloader.Downl
 
     @Override
     public void downloadStarted(int id, String fileName) {
-        notificationBuilder = new NotificationCompat.Builder(getApplicationContext(), DOWNLOAD_CHANNEL_ID);
+        notificationBuilder = new NotificationCompat.Builder(getApplicationContext(), AmahiApplication.DOWNLOAD_CHANNEL_ID);
         notificationBuilder
             .setOngoing(true)
             .setSmallIcon(R.drawable.ic_app_logo)

--- a/src/main/java/org/amahi/anywhere/service/DownloadService.java
+++ b/src/main/java/org/amahi/anywhere/service/DownloadService.java
@@ -1,10 +1,9 @@
 package org.amahi.anywhere.service;
 
-import android.app.ActivityManager;
+import android.annotation.SuppressLint;
 import android.app.DownloadManager;
 import android.app.Notification;
 import android.app.NotificationManager;
-import android.app.Service;
 import android.content.Context;
 import android.content.Intent;
 import android.net.ConnectivityManager;
@@ -46,10 +45,9 @@ import javax.inject.Inject;
 
 import static org.amahi.anywhere.util.Downloader.OFFLINE_PATH;
 
-public class DownloadService extends Service implements Downloader.DownloadCallbacks {
+public class DownloadService extends ServiceNotifier implements Downloader.DownloadCallbacks {
 
     private static final int NOTIFICATION_OFFLINE_ID = 101;
-    private static final int NOTIFICATION_ERROR_ID = 101;
     private static final String DOWNLOAD_CHANNEL_ID = "file_download";
 
     @Inject
@@ -76,6 +74,7 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
         setUpNetworkUtils();
         setDatabaseRepository();
         downloader.setDownloadCallbacks(this);
+        notificationBuilder = startForegroundNotif(DOWNLOAD_CHANNEL_ID);
         Log.i("DownloadService", "Download Service created");
     }
 
@@ -141,23 +140,8 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
             isDownloading = true;
         } else {
             stopDownloading();
-            if (isServiceRunningInForeground(this, DownloadService.class)) {
-                stopForeground(true);
-            }
+            stopForegroundService(this, this, true);
         }
-    }
-
-    public static boolean isServiceRunningInForeground(Context context, Class<?> serviceClass) {
-        ActivityManager manager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
-        for (ActivityManager.RunningServiceInfo service : manager.getRunningServices(Integer.MAX_VALUE)) {
-            if (serviceClass.getName().equals(service.service.getClassName())) {
-                if (service.foreground) {
-                    return true;
-                }
-
-            }
-        }
-        return false;
     }
 
     private void stopDownloading() {
@@ -168,14 +152,14 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
         offlineFile.setDownloadId(downloadId);
         offlineFileRepository.update(offlineFile);
     }
-    
+
     private void resumeDownload(long downloadId, String fileName) {
         notificationBuilder = new NotificationCompat.Builder(getApplicationContext(), DOWNLOAD_CHANNEL_ID);
         notificationBuilder
             .setOngoing(true)
             .setSmallIcon(R.drawable.ic_app_logo)
             .setContentTitle(getString(R.string.notification_download_offline_title))
-            .setContentText(getString(R.string.notification_upload_message, fileName))
+            .setContentText(fileName)
             .setProgress(100, 0, false)
             .build();
         Notification notification = notificationBuilder.build();
@@ -248,7 +232,7 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
             .setOngoing(true)
             .setSmallIcon(R.drawable.ic_app_logo)
             .setContentTitle(getString(R.string.notification_download_offline_title))
-            .setContentText(getString(R.string.notification_upload_message, fileName))
+            .setContentText(fileName)
             .setProgress(100, 0, false)
             .build();
         Notification notification = notificationBuilder.build();
@@ -267,7 +251,9 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
         notificationBuilder
             .setProgress(100, progress, false);
         Notification notification = notificationBuilder.build();
-        notificationManager.notify(NOTIFICATION_OFFLINE_ID, notification);
+        if (progress == 100) {
+            notificationManager.notify(NOTIFICATION_OFFLINE_ID, notification);
+        }
         updateOfflineActivityUI(progress);
     }
 
@@ -283,7 +269,7 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
         if (offlineFile != null) {
             moveFileInOfflineDirectory(offlineFile.getName());
         } else {
-            stopForeground(true);
+            stopForegroundService(this, this, true);
         }
     }
 
@@ -293,7 +279,7 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
 
         notificationBuilder
             .setContentTitle(getString(R.string.notification_offline_download_complete))
-            .setContentText(getString(R.string.notification_upload_message, offlineFile.getName()))
+            .setContentText(offlineFile.getName())
             .setOngoing(false);
 
         Notification notification = notificationBuilder.build();
@@ -315,7 +301,7 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
             offlineFile.setDownloadId(-1);
             offlineFileRepository.update(offlineFile);
 
-            stopForeground(true);
+            stopForegroundService(this, this, true);
 
             showDownloadedNotification(offlineFile);
             notifyDownloadFinish(true);
@@ -361,7 +347,7 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
 
             notificationBuilder
                 .setContentTitle(getString(R.string.notification_offline_download_error))
-                .setContentText(getString(R.string.notification_upload_message, offlineFile.getName()))
+                .setContentText(offlineFile.getName())
                 .setOngoing(false);
 
             Notification notification = notificationBuilder.build();
@@ -370,12 +356,12 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
     }
 
     private void stopNotification() {
-        stopForeground(true);
+        stopForegroundService(this, this, true);
     }
 
     @Override
     public void downloadPaused(long downloadId, int progress) {
-        stopForeground(false);
+        stopForegroundService(this, this, false);
         NotificationManager notificationManager = (NotificationManager) getApplicationContext()
             .getSystemService(Context.NOTIFICATION_SERVICE);
 
@@ -405,7 +391,7 @@ public class DownloadService extends Service implements Downloader.DownloadCallb
 
     @Subscribe
     public void onDownloadCanceled(OfflineCanceledEvent event) {
-        stopForeground(true);
+        stopForegroundService(this, this, true);
         List<OfflineFile> offlineFiles = offlineFileRepository.getAllOfflineFiles();
         if (offlineFiles.isEmpty()) {
             stopDownloading();

--- a/src/main/java/org/amahi/anywhere/service/ServiceNotifier.java
+++ b/src/main/java/org/amahi/anywhere/service/ServiceNotifier.java
@@ -37,7 +37,6 @@ abstract public class ServiceNotifier extends Service {
             .setSmallIcon(R.drawable.ic_app_logo)
             .setContentTitle(getString(R.string.application_name))
             .setShowWhen(false)
-            .setPriority(Notification.PRIORITY_MIN)
             .build();
         Notification notification = notificationBuilder.build();
         int nID = (int) System.currentTimeMillis() % 10000;

--- a/src/main/java/org/amahi/anywhere/service/ServiceNotifier.java
+++ b/src/main/java/org/amahi/anywhere/service/ServiceNotifier.java
@@ -1,0 +1,65 @@
+/*
+ * Copyright (c) 2014 Amahi
+ *
+ * This file is part of Amahi.
+ *
+ * Amahi is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Amahi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Amahi. If not, see <http ://www.gnu.org/licenses/>.
+ */
+
+package org.amahi.anywhere.service;
+
+import android.app.ActivityManager;
+import android.app.Notification;
+import android.app.Service;
+import android.content.Context;
+
+import androidx.core.app.NotificationCompat;
+
+import org.amahi.anywhere.R;
+
+abstract public class ServiceNotifier extends Service {
+
+    public NotificationCompat.Builder startForegroundNotif(String channelID) {
+        NotificationCompat.Builder notificationBuilder = new NotificationCompat.Builder(getApplicationContext(), channelID);
+        notificationBuilder
+            .setOngoing(true)
+            .setSmallIcon(R.drawable.ic_app_logo)
+            .setContentTitle(getString(R.string.application_name))
+            .setShowWhen(false)
+            .setPriority(Notification.PRIORITY_MIN)
+            .build();
+        Notification notification = notificationBuilder.build();
+        int nID = (int) System.currentTimeMillis() % 10000;
+        startForeground(nID, notification);
+        return notificationBuilder;
+    }
+
+    private static boolean isServiceRunningInForeground(Context context, Class<?> serviceClass) {
+        ActivityManager manager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
+        for (ActivityManager.RunningServiceInfo service : manager.getRunningServices(Integer.MAX_VALUE)) {
+            if (serviceClass.getName().equals(service.service.getClassName())) {
+                if (service.foreground) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public void stopForegroundService(Context ctx, Service serviceClass, boolean flag) {
+        if (isServiceRunningInForeground(ctx, serviceClass.getClass())) {
+            serviceClass.stopForeground(flag);
+        }
+    }
+}

--- a/src/main/java/org/amahi/anywhere/service/UploadService.java
+++ b/src/main/java/org/amahi/anywhere/service/UploadService.java
@@ -60,8 +60,6 @@ import javax.inject.Inject;
  */
 public class UploadService extends ServiceNotifier implements UploadManager.UploadCallbacks {
 
-    private static final String UPLOAD_CHANNEL_ID = "file_upload";
-
     @Inject
     ServerClient serverClient;
 
@@ -83,7 +81,7 @@ public class UploadService extends ServiceNotifier implements UploadManager.Uplo
         setUpBus();
         setUpDbHelper();
         setUpNetworkUtils();
-        startForegroundNotif(UPLOAD_CHANNEL_ID);
+        startForegroundNotif(AmahiApplication.UPLOAD_CHANNEL_ID);
     }
 
     private void setUpInjections() {
@@ -239,7 +237,7 @@ public class UploadService extends ServiceNotifier implements UploadManager.Uplo
 
     @Override
     public void uploadStarted(int id, String fileName) {
-        notificationBuilder = new NotificationCompat.Builder(getApplicationContext(), UPLOAD_CHANNEL_ID);
+        notificationBuilder = new NotificationCompat.Builder(getApplicationContext(), AmahiApplication.UPLOAD_CHANNEL_ID);
         notificationBuilder
             .setOngoing(true)
             .setSmallIcon(R.drawable.ic_app_logo)

--- a/src/main/java/org/amahi/anywhere/service/UploadService.java
+++ b/src/main/java/org/amahi/anywhere/service/UploadService.java
@@ -21,7 +21,6 @@ package org.amahi.anywhere.service;
 
 import android.app.Notification;
 import android.app.NotificationManager;
-import android.app.Service;
 import android.content.Context;
 import android.content.Intent;
 import android.content.SharedPreferences;
@@ -31,6 +30,7 @@ import android.os.Build;
 import android.os.IBinder;
 import android.preference.PreferenceManager;
 import android.provider.MediaStore;
+
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.core.app.NotificationCompat;
@@ -58,7 +58,7 @@ import javax.inject.Inject;
 /**
  * File upload service
  */
-public class UploadService extends Service implements UploadManager.UploadCallbacks {
+public class UploadService extends ServiceNotifier implements UploadManager.UploadCallbacks {
 
     private static final String UPLOAD_CHANNEL_ID = "file_upload";
 
@@ -83,6 +83,7 @@ public class UploadService extends Service implements UploadManager.UploadCallba
         setUpBus();
         setUpDbHelper();
         setUpNetworkUtils();
+        startForegroundNotif(UPLOAD_CHANNEL_ID);
     }
 
     private void setUpInjections() {
@@ -243,7 +244,7 @@ public class UploadService extends Service implements UploadManager.UploadCallba
             .setOngoing(true)
             .setSmallIcon(R.drawable.ic_app_logo)
             .setContentTitle(getString(R.string.notification_upload_title))
-            .setContentText(getString(R.string.notification_upload_message, fileName))
+            .setContentText(fileName)
             .setProgress(100, 0, false)
             .build();
         Notification notification = notificationBuilder.build();
@@ -271,7 +272,8 @@ public class UploadService extends Service implements UploadManager.UploadCallba
     }
 
     private void uploadComplete(int id, String title) {
-        stopForeground(false);
+        stopForegroundService(this, this, false);
+//        stopForeground(false);
         NotificationManager notificationManager = (NotificationManager) getApplicationContext()
             .getSystemService(Context.NOTIFICATION_SERVICE);
 

--- a/src/main/java/org/amahi/anywhere/service/UploadService.java
+++ b/src/main/java/org/amahi/anywhere/service/UploadService.java
@@ -185,6 +185,7 @@ public class UploadService extends ServiceNotifier implements UploadManager.Uplo
         SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
         String preferenceConnection = preferences.getString(getString(R.string.preference_key_server_connection), null);
 
+        assert preferenceConnection != null;
         return preferenceConnection.equals(getString(R.string.preference_key_server_connection_auto));
     }
 
@@ -192,6 +193,7 @@ public class UploadService extends ServiceNotifier implements UploadManager.Uplo
         SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
         String preferenceConnection = preferences.getString(getString(R.string.preference_key_server_connection), null);
 
+        assert preferenceConnection != null;
         return preferenceConnection.equals(getString(R.string.preference_key_server_connection_local));
     }
 
@@ -256,7 +258,9 @@ public class UploadService extends ServiceNotifier implements UploadManager.Uplo
         notificationBuilder
             .setProgress(100, progress, false);
         Notification notification = notificationBuilder.build();
-        notificationManager.notify(id, notification);
+        if (progress == 100) {
+            notificationManager.notify(id, notification);
+        }
     }
 
     @Override
@@ -270,16 +274,17 @@ public class UploadService extends ServiceNotifier implements UploadManager.Uplo
     }
 
     private void uploadComplete(int id, String title) {
-        stopForegroundService(this, this, false);
+        stopForegroundService(this, this, true);
         NotificationManager notificationManager = (NotificationManager) getApplicationContext()
             .getSystemService(Context.NOTIFICATION_SERVICE);
 
         notificationBuilder
             .setContentTitle(title)
-            .setOngoing(false)
-            .setProgress(0, 0, false);
+            .setContentText(getString(R.string.message_file_upload_complete))
+            .setOngoing(false);
 
         Notification notification = notificationBuilder.build();
+        notificationManager.cancel(id);
         notificationManager.notify(id, notification);
     }
 

--- a/src/main/java/org/amahi/anywhere/service/UploadService.java
+++ b/src/main/java/org/amahi/anywhere/service/UploadService.java
@@ -273,7 +273,6 @@ public class UploadService extends ServiceNotifier implements UploadManager.Uplo
 
     private void uploadComplete(int id, String title) {
         stopForegroundService(this, this, false);
-//        stopForeground(false);
         NotificationManager notificationManager = (NotificationManager) getApplicationContext()
             .getSystemService(Context.NOTIFICATION_SERVICE);
 

